File: students/students/src/MON.SPA/src/models/personalDevelopmentSupportModel.js
import { ReasonModel } from './reasonModel.js';

export class PersonalDevelopmentSupportModel {

    constructor(obj = {}) {
        this.id = obj.id || null,
        this.personId = obj.personId || null,
        this.schoolYear = obj.schoolYear || null,
        this.additionalModulesNeededForNonBulgarianSpeakingInfo = obj.additionalModulesNeededForNonBulgarianSpeakingInfo || '';
        this.earlyEvaluationAndEducationalRiskInfo = obj.earlyEvaluationAndEducationalRiskInfo || '';
        this.evaluationConclusionInfo = obj.evaluationConclusionInfo || '';
        this.studentTypeId = obj.studentTypeId || null;
        this.supportPeriodTypeId = obj.supportPeriodTypeId || null;
        this.earlyEvaluationReasons = obj.earlyEvaluationReasons ? obj.earlyEvaluationReasons.map(el => new ReasonModel(el)) : [];
        this.commonSupportTypeReasons = obj.commonSupportTypeReasons ? obj.commonSupportTypeReasons.map(el => new ReasonModel(el)) : [];
        this.additionalSupportTypeReasons = obj.additionalSupportTypeReasons ? obj.additionalSupportTypeReasons.map(el => new ReasonModel(el)) : [];
        this.earlyEvaluationDocuments = obj.earlyEvaluationDocuments || [];
        this.additionalSupportDocuments = obj.additionalSupportDocuments || [];
        this.commonSupportDocuments = obj.commonSupportDocuments || [];
    }
}

File: students/students/src/MON.Models/StudentModels/PersonalDevelopmentSupportModel.cs
using System.Collections.Generic;

namespace MON.Models.StudentModels
{
    public class PersonalDevelopmentSupportModel
    {
        public int? Id { get; set; }
        public int PersonId { get; set; }
        public short SchoolYear { get; set; }
        public string EarlyEvaluationAndEducationalRiskInfo { get; set; }
        public string AdditionalModulesNeededForNonBulgarianSpeakingInfo { get; set; }
        public string EvaluationConclusionInfo { get; set; }
        public int? SupportPeriodTypeId { get; set; }
        public int? StudentTypeId { get; set; }
        public IEnumerable<PersonalDevelopmenReasonsModel> EarlyEvaluationReasons { get; set; }
        public IEnumerable<PersonalDevelopmenReasonsModel> CommonSupportTypeReasons { get; set; }
        public IEnumerable<PersonalDevelopmenReasonsModel> AdditionalSupportTypeReasons { get; set; }
        public IEnumerable<DocumentModel> AdditionalSupportDocuments { get; set; }
        public IEnumerable<DocumentModel> CommonSupportDocuments { get; set; }
        public IEnumerable<DocumentModel> EarlyEvaluationDocuments { get; set; }
    }
}
